synthesizer/src/vm/mod.rs

```diff
@@ -335,12 +335,32 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
         // Attention: The following order is crucial because if 'finalize' fails, we can rollback the block.
         // If one first calls 'finalize', then calls 'insert(block)' and it fails, there is no way to rollback 'finalize'.
 
+        // Enable the atomic batch override, so that both the insertion and finalization belong to a single batch.
+        #[cfg(feature = "rocks")]
+        assert!(self.block_store().flip_atomic_override()?);
```

Comment:
Should this be `debug_assert!`?

Reply:
the plain `assert` was intentional, as this ensures that it is initially disabled - if this was not the case, it would indicate a logic bug